library/ssl_tls12_server.c

```diff
@@ -922,12 +922,15 @@ static int ssl_parse_client_hello(mbedtls_ssl_context *ssl)
      * If renegotiating, then the input was read with mbedtls_ssl_read_record(),
      * otherwise read it ourselves manually in order to support SSLv2
      * ClientHello, which doesn't use the same record layer format.
+     * Otherwise in a scenario of TLS 1.3/TLS 1.2 version negotiation, the
+     * ClientHello has been already fully fetched by the TLS 1.3 code and the
```

Comment:
Out of curiosity, is that flag being asserted by `ssl_tls13_check_new_session_ticket()` or can it be asserted by the client? 

If the client can set it, could it be used for a protocol downgrade?